### ISSUE TITLE
Can't import into Astro project

### ISSUE THREAD

**Issue by MystPi** [radix-ui/themes#13]
This is my Astro layout:

```astro
---
import '@radix-ui/themes/styles.css';
import { Theme } from '@radix-ui/themes';

interface Props {
  title: string;
}

const { title } = Astro.props;
---

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="description" content="Astro description" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
    <meta name="generator" content={Astro.generator} />
    <title>{title}</title>
  </head>
  <body>
		<Theme>
			<slot />
		</Theme>
  </body>
</html>
```

However, I can't import `Theme` from `@radix-ui/themes` because of this error:
```
 error   Unexpected token 'export'
/home/.../node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1176:20)
    at Module._compile (node:internal/modules/cjs/loader:1218:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25) (x2)
```
Any help would be appreciated.

**Comment by milesj** [comment]
Running into this also.

**Comment by milesj** [comment]
Now that Astro 3 is out, it would be great if this worked.

**Comment by kiastorm** [comment]
Getting this also

```
Error when evaluating SSR module /src/components/test.tsx: failed to import "@radix-ui/themes"
|- /Users/kiastorm/Documents/dev/nebulous-neutron/node_modules/.pnpm/@radix-ui+themes@1.1.2_@types+react-dom@18.2.7_@types+react@18.2.21_react-dom@18.2.0_react@18.2.0/node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

10:41:15 AM [vite] Error when evaluating SSR module /Users/kiastorm/Documents/dev/nebulous-neutron/src/pages/index.astro: failed to import "/src/components/test.tsx"
|- /Users/kiastorm/Documents/dev/nebulous-neutron/node_modules/.pnpm/@radix-ui+themes@1.1.2_@types+react-dom@18.2.7_@types+react@18.2.21_react-dom@18.2.0_react@18.2.0/node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

 error   Unexpected token 'export'
/Users/kiastorm/Documents/dev/nebulous-neutron/node_modules/.pnpm/@radix-ui+themes@1.1.2_@types+react-dom@18.2.7_@types+react@18.2.21_react-dom@18.2.0_react@18.2.0/node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
 error   Unexpected token 'export'
/Users/kiastorm/Documents/dev/nebulous-neutron/node_modules/.pnpm/@radix-ui+themes@1.1.2_@types+react-dom@18.2.7_@types+react@18.2.21_react-dom@18.2.0_react@18.2.0/node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25) (x2)
```

**Comment by kiastorm** [comment]
Worked around this by adding the package to `vite.srr.noExternal`. However, now I'm getting "Sourcemap points to missing source files" warnings. It builds and renders successfully though!

```
// astro.config.mjs

export default defineConfig({
  integrations: [react()],
  vite: {
    ssr: {
      noExternal: ["@radix-ui/themes"],
    },
  },
});
```

**Comment by o-az** [comment]
> Worked around this by adding the package to `vite.srr.noExternal`. However, now I'm getting "Sourcemap points to missing source files" warnings. It builds and renders successfully though!
> 
> ```
> // astro.config.mjs
> 
> export default defineConfig({
>   integrations: [react()],
>   vite: {
>     ssr: {
>       noExternal: ["@radix-ui/themes"],
>     },
>   },
> });
> ```

Same thing here: 
![CleanShot 2023-09-05 at 11 07 51@2x](https://github.com/radix-ui/themes/assets/23618431/c27cd576-1abd-4408-b095-9d6ede6cd18d)

**Comment by AlexEscalante** [comment]
I can also report this issue, with a newly created Astro 3 project:

`6:18:21 PM [vite] Error when evaluating SSR module /src/layouts/Layout.astro: failed to import "@radix-ui/themes"
|- /Users/alex/Devel/semillero/web/node_modules/@radix-ui/themes/dist/esm/index.js:1
export { Theme, useThemeContext, updateThemeAppearanceClass } from './theme';
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:74:18)
    at wrapSafe (node:internal/modules/cjs/loader:1141:20)
    at Module._compile (node:internal/modules/cjs/loader:1182:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
`

**Comment by milesj** [comment]
It kind of seems like vite thinks the files are CJS, even though they are importing the ESM files.

**Comment by mujtabachang** [comment]
Same issue here. Would be great is someone could find a fix to it

**Comment by KubaJastrz** [comment]
Probably because of this issue reported by Publint:

> ./dist/esm/index.js is written in ESM, but is interpreted as CJS. Consider using the .mjs extension, e.g. ./dist/esm/index.mjs

https://publint.dev/@radix-ui/themes@2.0.0

**Comment by fjuren** [comment]
Ran into this one today. Will comment if I learn anything more about the issue.

**Comment by vladmoroz** [comment]
This will be fixed in the upcoming 3.0.0 version.

We released `"@radix-ui/themes": "3.0.0-rc.1"` which should work in ESM projects if you need an early version now. Please refer to the changelog for other changes in this release.
